src/java.base/share/native/launcher/main.c

```diff
@@ -278,15 +321,32 @@ main(int argc, char **argv)
         margv = args->elements;
     }
 
-    // Avoid unexpected process completion when checkpointing under docker container run
-    // by creating the main process waiting for children before exit.
-    if (is_checkpoint && 1 == getpid()) {
-        g_child_pid = fork();
-        if (0 < g_child_pid) {
-            // The main process should forward signals to the child.
-            setup_sighandler();
-            const int status = wait_for_children();
-            exit(status);
+    if (is_checkpoint) {
+        const int crac_min_pid_default = 128;
+        const char *env_min_pid_str = getenv("CRAC_MIN_PID");
+        const int env_min_pid = env_min_pid_str ? atoi(env_min_pid_str) : 0;
+        // TODO: should it be checked for max pid overflow?
+        const int crac_min_pid = 0 < env_min_pid ? env_min_pid : crac_min_pid_default;
+
+        if (getpid() <= crac_min_pid) {
```

Comment:
This should work for pid==1 as well. 
The possible values for crac_min_pid are 1, 2, 3, ...
In case of `getpid() < crac_min_pid`, it needs to have an additional check.

Reply:
As we discussed this offline, now PID adjustment is not applied by default, it works only if CracMinPid option is set by an user. So 'fork & wait' now is applied for pid==1 or for PID adjustment.